tools/verify-fs.in

```diff
@@ -95,25 +95,36 @@ for ROOTFS in *.squashfs; do
     fi
 
     if [ -s "${ROOTFS}.sha256" ] ; then
-        if [ /usr/bin/sha256sum -c "${ROOTFS}.sha256" 2>/dev/null ] ; then
-            log "Good integrity of ${ROOTFS}, SHA256 checksum is valid"
+        if /usr/bin/sha256sum -c "${ROOTFS}.sha256" 2>/dev/null; then
+            if [ -s "${ROOTFS}-manifest.json" ] ; then
+                    SHA256_COMPUTED=$(cut -d ' ' -f1 < "${ROOTFS}.sha256")
+                    SHA256_MANIFEST=$(/usr/share/fty/scripts/JSON.sh -l < "${ROOTFS}-manifest.json" | grep '\["application","ShaChecksum"\]' | cut -f2 | cut -d '"' -f2)
+
+                if [ "${SHA256_COMPUTED}" = "${SHA256_MANIFEST}" ]; then
+                    if [ -s "${ROOTFS}-manifest.json.p7s" ] ; then
+                        SIGNED_BY=$(/usr/share/fty/scripts/JSON.sh -l < "${ROOTFS}-manifest.json" | grep '\["application","SignedBy"\]' | cut -f2 | cut -d '"' -f2)
+
+                        if openssl cms -verify -binary -inform der -CAfile "/usr/share/authorities/${SIGNED_BY}/rootCa1.pem" -purpose any -out /dev/null -content "${ROOTFS}-manifest.json" -in "${ROOTFS}-manifest.json.p7s"; then
+                            log "Image signature is valid for ${ROOTFS}"
+                        else
+                            log "Image signature is invalid for ${ROOTFS}"
+                        fi
+                    else
+                        log "Manifest signature not found for ${ROOTFS}"
+                    fi
+                else
+                    log "Manifest sha256 hash does not match with the sha256 hash for ${ROOTFS}"
+                fi
+            else
+                log "Manifest not found for ${ROOTFS}"
+            fi
         else
             log "Bad integrity of ${ROOTFS}, SHA256 checksum is not valid"
         fi
     else
         log "Missing SHA256 checksum file: '${ROOTFS}.sha256'"
     fi
 
-    if [ -s "${ROOTFS}.cksum" ] ; then
-        if [ /usr/bin/cksum -c "${ROOTFS}.cksum" 2>/dev/null ] ; then
```

Comment:
Note: removing this block misses out on size-verification of cksum (we might also use the size embedded to manifest).

But notably our images have a version of this tool that does not support `cksum -c` so this block would need at least fixing to pass like done in other places (call `cksum FILENAME` and pick first two columns of output, compare to the two in saved cksum file).